crates/astria-sequencer-client/src/lib.rs

```diff
@@ -248,60 +197,54 @@ impl ErrorKind {
             rpc,
         })
     }
-
-    /// Convenience method to construct a `WrongQueryResponseKind` variant.
-    fn wrong_query_response_kind(
-        expected: &'static str,
-        received: sequencer::accounts::query::Response,
-    ) -> Self {
-        Self::WrongQueryResponseKind(WrongQueryResponseKind {
-            expected,
-            received,
-        })
-    }
 }
 
 /// Tendermint HTTP client which is used to interact with the Sequencer node.
 #[async_trait]
 pub trait SequencerClientExt: Client {
     /// Returns the balance of the given account at the given height.
     ///
-    /// If `height = None`, the latest height is used.
-    ///
     /// # Errors
     ///
     /// - If calling tendermint `abci_query` RPC fails.
-    /// - If the opaque bytes cannot be deserialized as a sequencer query response.
-    /// - If the deserialized sequencer query response is not a balance response.
-    async fn get_balance(
-        &self,
-        address: &Address,
-        height: Option<Height>,
-    ) -> Result<Balance, Error> {
-        use borsh::BorshDeserialize as _;
-        use sequencer::accounts::query::Response;
-        let height = height.map(Into::into);
+    /// - If the bytes contained in the abci query response cannot be read as an
+    ///   `astria.sequencer.v1alpha1.BalanceResponse`.
+    async fn get_balance(&self, address: [u8; 20], height: u32) -> Result<BalanceResponse, Error> {
+        use proto::Message as _;
+
+        let prefix = b"accounts/balance/";
+        let mut path = vec![0u8; prefix.len() + 20 * 2];
+        path.extend_from_slice(prefix);
+        faster_hex::hex_encode(&address, &mut path).expect(
```

Comment:
what's the benefit of the `faster_hex` library? also should `&mut path` be `&mut path[prefix.len()..]` since the address goes after the prefix?

Reply:
I removed `faster_hex` because it's orthogonal to this PR. In general it's better to use it over `hex` because it's a lot faster.

Re the path: tysm! I messed up while making the tests work, and it turns out the mock tests didn't catch thits. Added a helper function to the construct to make sure it lines up with what we expect.

In general the mock tests in this crate should probably be replaced by integration tests against an actual sequencer (+ cometbft).